File: app/src/main/java/com/appsys/android/popularmovie/classes/MovieTrailer.java
package com.appsys.android.popularmovie.classes;

import android.net.Uri;
import android.os.Parcel;
import android.os.Parcelable;

import com.appsys.android.popularmovie.api.TheMovieDbApi;

/**
 * Created by shakir on 8/8/2017.
 */

public class MovieTrailer implements Parcelable {

    private static final String TAG = MovieTrailer.class.getSimpleName();

    private int mMovieId;
    private String mKey;
    private String mTitle;
    private String mSite;
    private String mType;

    public MovieTrailer(int movieId, String key, String title, String site, String type) {
        mMovieId = movieId;
        mKey = key;
        mTitle = title;
        mSite = site;
        mType = type;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeInt(mMovieId);
        parcel.writeString(mKey);
        parcel.writeString(mTitle);
        parcel.writeString(mSite);
        parcel.writeString(mType);
    }

    public final static Parcelable.Creator<MovieTrailer> CREATOR = new Parcelable.Creator<MovieTrailer>() {
        @Override
        public MovieTrailer createFromParcel(Parcel parcel) {
            return new MovieTrailer(parcel);
        }

        @Override
        public MovieTrailer[] newArray(int i) {
            return new MovieTrailer[0];
        }
    };

    public MovieTrailer(Parcel parcel) {
        mMovieId = parcel.readInt();
        mKey = parcel.readString();
        mTitle = parcel.readString();
        mSite = parcel.readString();
        mType = parcel.readString();
    }

    public int getMovieId() {
        return mMovieId;
    }

    public String getTitle() {
        return mTitle;
    }

    public Uri getImageUrl() {
        return TheMovieDbApi.getYouTubeImagePath(mKey);
    }

    public Uri getVideoUrl() {
        return TheMovieDbApi.getYouTubeVideoPath(mKey);
    }
}

File: app/src/main/java/com/appsys/android/popularmovie/classes/MovieTrailer.kt
package com.appsys.android.popularmovie.classes

import android.net.Uri
import android.os.Parcel
import android.os.Parcelable
import com.appsys.android.popularmovie.api.TheMovieDbApi.Companion.getYouTubeImagePath
import com.appsys.android.popularmovie.api.TheMovieDbApi.Companion.getYouTubeVideoPath

/**
 * Created by shakir on 8/8/2017.
 */
class MovieTrailer(val movieId: Int, val key: String?, val title: String?, val site: String?, val type: String?) : Parcelable {

    constructor(parcel: Parcel) : this(parcel.readInt(), parcel.readString(), parcel.readString(), parcel.readString(), parcel.readString())

    override fun describeContents() = 0

    override fun writeToParcel(parcel: Parcel, i: Int) {
        parcel.writeInt(movieId)
        parcel.writeString(key)
        parcel.writeString(title)
        parcel.writeString(site)
        parcel.writeString(type)
    }

    val imageUrl: Uri
        get() = getYouTubeImagePath(key)

    val videoUrl: Uri
        get() = getYouTubeVideoPath(key)

    companion object CREATOR : Parcelable.Creator<MovieTrailer> {
        override fun createFromParcel(parcel: Parcel) = MovieTrailer(parcel)

        override fun newArray(i: Int) = arrayOfNulls<MovieTrailer>(i)
    }
}